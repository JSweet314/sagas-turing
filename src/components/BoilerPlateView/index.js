import React from 'react';
import {func, string, shape } from 'prop-types';

export const BoilerPlateView = ({
  handleOnClick,
  handleOnChange,
  inputValue,
  user,
  error
}) => (
  <div>
    <h1>React/Redux Boilerplate</h1>
    <p>Includes:</p>
    <ul>
      <a
        href="https://github.com/reduxjs/redux"
        target="_blank"
        rel="noopener noreferrer">
        <li>Redux</li>
      </a>
      <a
        href="https://github.com/redux-saga/redux-saga"
        target="_blank"
        rel="noopener noreferrer">
        <li>Redux-Saga Middleware</li>
      </a>
      <li>
        Testing w/{' '}
        <a
          href="https://github.com/facebook/jest"
          target="_blank"
          rel="noopener noreferrer">
          Jest
        </a>{' '}
        +{' '}
        <a
          href="https://github.com/airbnb/enzyme"
          target="_blank"
          rel="noopener noreferrer">
          Enzyme
        </a>
      </li>
    </ul>
    <p>
      This app is bootstrapped on top of{' '}
      <a
        href="https://github.com/facebook/create-react-app"
        target="_blank"
        rel="noopener noreferrer">
        Create-React-App
      </a>
    </p>
    <input
      value={inputValue}
      onChange={event => handleOnChange(event)}
      placeholder="GitHub Username"
      name="inputValue"
    />
    <button onClick={event => handleOnClick(event)}>Test Middleware</button>
    <p>{error || user.login}</p>
  </div>
);

BoilerPlateView.propTypes = {
  handleOnClick: func.isRequired,
  handleOnChange: func.isRequired,
  user: shape({
    login: string
  }).isRequired,
  error: string
};

export default BoilerPlateView;
