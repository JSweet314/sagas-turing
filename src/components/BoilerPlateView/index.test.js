import React from 'react';
import BoilerPlateView from './index';
import { shallow } from 'enzyme';

describe('BoilerPlateView', () => {
  let wrapper, mockError, mockUser, mockHandleOnClick;

  beforeEach(() => {
    mockHandleOnClick = jest.fn();
    mockUser = {
      login: 'jsweet314'
    };
    mockError = 'Error - GitHub User Not Found';
    wrapper = shallow(
      <BoilerPlateView
        handleOnClick={mockHandleOnClick}
        user={mockUser}
        error={mockError}
      />
    );
  });

  it('should match a snapshot', () => {
    expect(wrapper).toMatchSnapshot();
  });
});
