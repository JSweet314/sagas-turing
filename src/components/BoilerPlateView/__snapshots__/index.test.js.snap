// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BoilerPlateView should match a snapshot 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Unknown
    error="Error - GitHub User Not Found"
    handleOnChange={[Function]}
    handleOnClick={[Function]}
    user={
        Object {
            "login": "jsweet314",
          }
    }
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <h1>
          React/Redux Boilerplate
</h1>,
        <p>
          Includes:
</p>,
        <ul>
          <a
                    href="https://github.com/reduxjs/redux"
                    rel="noopener noreferrer"
                    target="_blank"
          >
                    <li>
                              Redux
                    </li>
          </a>
          <a
                    href="https://github.com/redux-saga/redux-saga"
                    rel="noopener noreferrer"
                    target="_blank"
          >
                    <li>
                              Redux-Saga Middleware
                    </li>
          </a>
          <li>
                    Testing w/
                     
                    <a
                              href="https://github.com/facebook/jest"
                              rel="noopener noreferrer"
                              target="_blank"
                    >
                              Jest
                    </a>
                     
                    +
                     
                    <a
                              href="https://github.com/airbnb/enzyme"
                              rel="noopener noreferrer"
                              target="_blank"
                    >
                              Enzyme
                    </a>
          </li>
</ul>,
        <p>
          This app is bootstrapped on top of
           
          <a
                    href="https://github.com/facebook/create-react-app"
                    rel="noopener noreferrer"
                    target="_blank"
          >
                    Create-React-App
          </a>
</p>,
        <input
          name="inputValue"
          onChange={[Function]}
          placeholder="GitHub Username"
          value={undefined}
/>,
        <button
          onClick={[Function]}
>
          Test Middleware
</button>,
        <p>
          Error - GitHub User Not Found
</p>,
      ],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "React/Redux Boilerplate",
        },
        "ref": null,
        "rendered": "React/Redux Boilerplate",
        "type": "h1",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "Includes:",
        },
        "ref": null,
        "rendered": "Includes:",
        "type": "p",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <a
              href="https://github.com/reduxjs/redux"
              rel="noopener noreferrer"
              target="_blank"
>
              <li>
                            Redux
              </li>
</a>,
            <a
              href="https://github.com/redux-saga/redux-saga"
              rel="noopener noreferrer"
              target="_blank"
>
              <li>
                            Redux-Saga Middleware
              </li>
</a>,
            <li>
              Testing w/
               
              <a
                            href="https://github.com/facebook/jest"
                            rel="noopener noreferrer"
                            target="_blank"
              >
                            Jest
              </a>
               
              +
               
              <a
                            href="https://github.com/airbnb/enzyme"
                            rel="noopener noreferrer"
                            target="_blank"
              >
                            Enzyme
              </a>
</li>,
          ],
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": <li>
                Redux
</li>,
              "href": "https://github.com/reduxjs/redux",
              "rel": "noopener noreferrer",
              "target": "_blank",
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "Redux",
              },
              "ref": null,
              "rendered": "Redux",
              "type": "li",
            },
            "type": "a",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": <li>
                Redux-Saga Middleware
</li>,
              "href": "https://github.com/redux-saga/redux-saga",
              "rel": "noopener noreferrer",
              "target": "_blank",
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "Redux-Saga Middleware",
              },
              "ref": null,
              "rendered": "Redux-Saga Middleware",
              "type": "li",
            },
            "type": "a",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                "Testing w/",
                " ",
                <a
                  href="https://github.com/facebook/jest"
                  rel="noopener noreferrer"
                  target="_blank"
>
                  Jest
</a>,
                " ",
                "+",
                " ",
                <a
                  href="https://github.com/airbnb/enzyme"
                  rel="noopener noreferrer"
                  target="_blank"
>
                  Enzyme
</a>,
              ],
            },
            "ref": null,
            "rendered": Array [
              "Testing w/",
              " ",
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "Jest",
                  "href": "https://github.com/facebook/jest",
                  "rel": "noopener noreferrer",
                  "target": "_blank",
                },
                "ref": null,
                "rendered": "Jest",
                "type": "a",
              },
              " ",
              "+",
              " ",
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "Enzyme",
                  "href": "https://github.com/airbnb/enzyme",
                  "rel": "noopener noreferrer",
                  "target": "_blank",
                },
                "ref": null,
                "rendered": "Enzyme",
                "type": "a",
              },
            ],
            "type": "li",
          },
        ],
        "type": "ul",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            "This app is bootstrapped on top of",
            " ",
            <a
              href="https://github.com/facebook/create-react-app"
              rel="noopener noreferrer"
              target="_blank"
>
              Create-React-App
</a>,
          ],
        },
        "ref": null,
        "rendered": Array [
          "This app is bootstrapped on top of",
          " ",
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "Create-React-App",
              "href": "https://github.com/facebook/create-react-app",
              "rel": "noopener noreferrer",
              "target": "_blank",
            },
            "ref": null,
            "rendered": "Create-React-App",
            "type": "a",
          },
        ],
        "type": "p",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "name": "inputValue",
          "onChange": [Function],
          "placeholder": "GitHub Username",
          "value": undefined,
        },
        "ref": null,
        "rendered": null,
        "type": "input",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "Test Middleware",
          "onClick": [Function],
        },
        "ref": null,
        "rendered": "Test Middleware",
        "type": "button",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "Error - GitHub User Not Found",
        },
        "ref": null,
        "rendered": "Error - GitHub User Not Found",
        "type": "p",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <h1>
            React/Redux Boilerplate
</h1>,
          <p>
            Includes:
</p>,
          <ul>
            <a
                        href="https://github.com/reduxjs/redux"
                        rel="noopener noreferrer"
                        target="_blank"
            >
                        <li>
                                    Redux
                        </li>
            </a>
            <a
                        href="https://github.com/redux-saga/redux-saga"
                        rel="noopener noreferrer"
                        target="_blank"
            >
                        <li>
                                    Redux-Saga Middleware
                        </li>
            </a>
            <li>
                        Testing w/
                         
                        <a
                                    href="https://github.com/facebook/jest"
                                    rel="noopener noreferrer"
                                    target="_blank"
                        >
                                    Jest
                        </a>
                         
                        +
                         
                        <a
                                    href="https://github.com/airbnb/enzyme"
                                    rel="noopener noreferrer"
                                    target="_blank"
                        >
                                    Enzyme
                        </a>
            </li>
</ul>,
          <p>
            This app is bootstrapped on top of
             
            <a
                        href="https://github.com/facebook/create-react-app"
                        rel="noopener noreferrer"
                        target="_blank"
            >
                        Create-React-App
            </a>
</p>,
          <input
            name="inputValue"
            onChange={[Function]}
            placeholder="GitHub Username"
            value={undefined}
/>,
          <button
            onClick={[Function]}
>
            Test Middleware
</button>,
          <p>
            Error - GitHub User Not Found
</p>,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "React/Redux Boilerplate",
          },
          "ref": null,
          "rendered": "React/Redux Boilerplate",
          "type": "h1",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Includes:",
          },
          "ref": null,
          "rendered": "Includes:",
          "type": "p",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <a
                href="https://github.com/reduxjs/redux"
                rel="noopener noreferrer"
                target="_blank"
>
                <li>
                                Redux
                </li>
</a>,
              <a
                href="https://github.com/redux-saga/redux-saga"
                rel="noopener noreferrer"
                target="_blank"
>
                <li>
                                Redux-Saga Middleware
                </li>
</a>,
              <li>
                Testing w/
                 
                <a
                                href="https://github.com/facebook/jest"
                                rel="noopener noreferrer"
                                target="_blank"
                >
                                Jest
                </a>
                 
                +
                 
                <a
                                href="https://github.com/airbnb/enzyme"
                                rel="noopener noreferrer"
                                target="_blank"
                >
                                Enzyme
                </a>
</li>,
            ],
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": <li>
                  Redux
</li>,
                "href": "https://github.com/reduxjs/redux",
                "rel": "noopener noreferrer",
                "target": "_blank",
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "Redux",
                },
                "ref": null,
                "rendered": "Redux",
                "type": "li",
              },
              "type": "a",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": <li>
                  Redux-Saga Middleware
</li>,
                "href": "https://github.com/redux-saga/redux-saga",
                "rel": "noopener noreferrer",
                "target": "_blank",
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "Redux-Saga Middleware",
                },
                "ref": null,
                "rendered": "Redux-Saga Middleware",
                "type": "li",
              },
              "type": "a",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  "Testing w/",
                  " ",
                  <a
                    href="https://github.com/facebook/jest"
                    rel="noopener noreferrer"
                    target="_blank"
>
                    Jest
</a>,
                  " ",
                  "+",
                  " ",
                  <a
                    href="https://github.com/airbnb/enzyme"
                    rel="noopener noreferrer"
                    target="_blank"
>
                    Enzyme
</a>,
                ],
              },
              "ref": null,
              "rendered": Array [
                "Testing w/",
                " ",
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": "Jest",
                    "href": "https://github.com/facebook/jest",
                    "rel": "noopener noreferrer",
                    "target": "_blank",
                  },
                  "ref": null,
                  "rendered": "Jest",
                  "type": "a",
                },
                " ",
                "+",
                " ",
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": "Enzyme",
                    "href": "https://github.com/airbnb/enzyme",
                    "rel": "noopener noreferrer",
                    "target": "_blank",
                  },
                  "ref": null,
                  "rendered": "Enzyme",
                  "type": "a",
                },
              ],
              "type": "li",
            },
          ],
          "type": "ul",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              "This app is bootstrapped on top of",
              " ",
              <a
                href="https://github.com/facebook/create-react-app"
                rel="noopener noreferrer"
                target="_blank"
>
                Create-React-App
</a>,
            ],
          },
          "ref": null,
          "rendered": Array [
            "This app is bootstrapped on top of",
            " ",
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "Create-React-App",
                "href": "https://github.com/facebook/create-react-app",
                "rel": "noopener noreferrer",
                "target": "_blank",
              },
              "ref": null,
              "rendered": "Create-React-App",
              "type": "a",
            },
          ],
          "type": "p",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "name": "inputValue",
            "onChange": [Function],
            "placeholder": "GitHub Username",
            "value": undefined,
          },
          "ref": null,
          "rendered": null,
          "type": "input",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Test Middleware",
            "onClick": [Function],
          },
          "ref": null,
          "rendered": "Test Middleware",
          "type": "button",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Error - GitHub User Not Found",
          },
          "ref": null,
          "rendered": "Error - GitHub User Not Found",
          "type": "p",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
    "attachTo": undefined,
    "hydrateIn": undefined,
  },
}
`;
